nncf/quantization/algorithms/min_max/torch_backend.py

```diff
@@ -0,0 +1,288 @@
+from nncf.torch.nncf_network import NNCFNetwork
+from nncf.torch.nncf_network import PTInsertionPoint
+from nncf.torch.graph.graph import PTTargetPoint
+from nncf.torch.graph.transformations.commands import PTInsertionCommand
+from nncf.torch.tensor_statistics.collectors import PTMinMaxStatisticCollector
+from nncf.torch.tensor_statistics.collectors import PTMeanMinMaxStatisticCollector
+from nncf.torch.quantization.layers import QUANTIZATION_MODULES
+from nncf.torch.quantization.layers import PTQuantizerSpec
+from nncf.torch.quantization.layers import get_scale_shape
+from nncf.torch.tensor_statistics.statistics import PTMinMaxTensorStatistic
+from nncf.torch.quantization.init_range import PTRangeInitCollectorParams
+from nncf.torch.quantization.init_range import StatCollectorGenerator
+from nncf.common.quantization.initialization.range import RangeInitConfig
+
+from nncf.quantization.algorithms.min_max.backend import MinMaxAlgoBackend
+from nncf.quantization.algorithms.min_max.backend import ALGO_BACKENDS
+
+import nncf.torch.graph.operator_metatypes as om
+
+@ALGO_BACKENDS.register(BackendType.TORCH)
+class PTMinMaxAlgoBackend(MinMaxAlgoBackend):
+
+    @property
+    def layers_with_weights_metatypes(self) -> List[OperatorMetatype]:
+        return [
+            om.PTModuleConv1dMetatype,
+            om.PTModuleConv2dMetatype,
+            om.PTModuleConv3dMetatype,
+            om.PTDepthwiseConv1dSubtype,
+            om.PTDepthwiseConv2dSubtype,
+            om.PTDepthwiseConv3dSubtype,
+            om.PTModuleLinearMetatype,
+            om.PTModuleConvTranspose1dMetatype,
+            om.PTModuleConvTranspose2dMetatype,
+            om.PTModuleConvTranspose3dMetatype,
+            om.PTModuleEmbeddingMetatype,
+            om.PTModuleEmbeddingBagMetatype,
+        ]
+
+    @property
+    def post_processing_metatypes(self) -> List[OperatorMetatype]:
+        return None
+
+    @property
+    def hw_fused_patterns(self) -> HWFusedPatterns:
+        return PT_HW_FUSED_PATTERNS
+
+    @property
+    def hw_config(self) -> HWConfig:
+        return PTHWConfig
+
+    @property
+    def quant_trait_op_dict(self) -> Dict[int, OperatorMetatype]:
+        return DEFAULT_PT_QUANT_TRAIT_TO_OP_DICT
+
+    @staticmethod
+    def model_transformer(model: NNCFNetwork) -> ModelTransformer:
+        return PTModelTransformer(model)
+
+    TARGET_TYPE_TO_PT_INS_TYPE_MAP = {
+        TargetType.PRE_LAYER_OPERATION: TargetType.OPERATOR_PRE_HOOK,
+        TargetType.POST_LAYER_OPERATION: TargetType.OPERATOR_POST_HOOK,
+    }
+
+    @staticmethod
+    def target_point(target_type: TargetType,
+                     target_node_name: str,
+                     port_id: int) -> PTTargetPoint:
+        if NNCFGraphNodeType.INPUT_NODE in target_node_name:
+            port_id = None
+        if target_type in PTMinMaxAlgoBackend.TARGET_TYPE_TO_PT_INS_TYPE_MAP:
+            target_type = PTMinMaxAlgoBackend.TARGET_TYPE_TO_PT_INS_TYPE_MAP[target_type]
```

Comment:
This translation is needed to valid transformation commands for `PTModelTransformer`